flink-ml-lib/src/main/java/org/apache/flink/ml/classification/linear/LogisticRegressionModel.java

```diff
@@ -0,0 +1,184 @@
+        implements Model<LogisticRegressionModel>,
+                LogisticRegressionModelParams<LogisticRegressionModel> {
+
+    private Map<Param<?>, Object> paramMap = new HashMap<>();
+
+    private Table modelData;
+
+    public LogisticRegressionModel() {
+        ParamUtils.initializeMapWithDefaultValues(this.paramMap, this);
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return paramMap;
+    }
+
+    @Override
+    public void save(String path) throws IOException {
+        ReadWriteUtils.saveMetadata(this, path);
+        ReadWriteUtils.saveModelData(
+                LogisticRegressionModelData.getModelDataStream(modelData),
+                path,
+                LogisticRegressionModelData.getModelDataEncoder());
+    }
+
+    public static LogisticRegressionModel load(StreamExecutionEnvironment env, String path)
+            throws IOException {
+        LogisticRegressionModel model = ReadWriteUtils.loadStageParam(path);
+        Table modelData =
+                ReadWriteUtils.loadModelData(
+                        env, path, LogisticRegressionModelData.getModelDataDecoder());
+        return model.setModelData(modelData);
+    }
+
+    @Override
+    public LogisticRegressionModel setModelData(Table... inputs) {
+        modelData = inputs[0];
+        return this;
+    }
+
+    @Override
+    public Table[] getModelData() {
+        return new Table[] {modelData};
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Table[] transform(Table... inputs) {
+        Preconditions.checkArgument(inputs.length == 1);
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+        DataStream<Row> inputStream = tEnv.toDataStream(inputs[0]);
+        final String broadcastModelKey = "broadcastModelKey";
+        DataStream<LogisticRegressionModelData> modelData =
+                LogisticRegressionModelData.getModelDataStream(this.modelData);
+        RowTypeInfo inputTypeInfo = TableUtils.getRowTypeInfo(inputs[0].getResolvedSchema());
+        RowTypeInfo outputTypeInfo =
+                new RowTypeInfo(
+                        ArrayUtils.addAll(
+                                inputTypeInfo.getFieldTypes(),
+                                BasicTypeInfo.DOUBLE_TYPE_INFO,
+                                TypeInformation.of(DenseVector.class)),
+                        ArrayUtils.addAll(
+                                inputTypeInfo.getFieldNames(),
+                                getPredictionCol(),
+                                getRawPredictionCol()));
+        DataStream<Row> predictionResult =
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(inputStream),
+                        Collections.singletonMap(broadcastModelKey, modelData),
+                        inputList -> {
+                            DataStream inputData = inputList.get(0);
```

Comment:
It would be nice to let `BroadcastUtils.withBroadcastStream` take `DataStreamList` instead of `List<DataStream<?>>`. This would our code style more consistent (i.e. use `DataStreamList` where it works). And then `DataStream inputData` could have the element type here.

Reply:
Hmm I think this should be a nice improvement. But should we do it in another PR?

Reply:
Sure, it's update to you. Feel free to do it in another PR.